src/components/views/donate/CryptoDonation.tsx

```diff
@@ -80,6 +80,8 @@ const CryptoDonation: FC = () => {
 		isConnected,
 		balance,
 	} = useGeneralWallet();
+	const { connection: solanaConnection } = useConnection();
+	const { publicKey } = useWallet();
```

Comment:
The address passed from `useGeneralWallet` has the same value, we don't need to define this one.

Reply:
so, what's your suggestion?
The "walletAddress" passed "useGeneralWallet" has the wallet address of the current user in string format, but I don't need the wallet address. I need the publicKey of the SPL token to fetch user token account related to user selected token to fetch its balance.
We need to check token balance before allowing user to donate.

Reply:
Done! I changed the code to use "useGeneralWallet" instead of "useWallet" but we still need to convert it.
The walletAddress returned from "useGeneralWallet" is in Base58 format but here I need it in PublicKey format.